packages/openapi-to-graphql/docs/SUBSCRIPTIONS.md

```diff
@@ -0,0 +1,279 @@
+# Subscriptions with OpenAPI-to-GraphQL
+Since version x.x.x, OpenAPI-to-GraphQL supports [GraphQL _subscription_ operations](http://spec.graphql.org/draft/#sec-Subscription). In GraphQL, using a subscription query, clients subscribe to updates on the data defined in the query. In this scneario, when data changes, the server publishes these changes to all clients that have active subscriptions for that data.
+
+The OpenAPI specification can define similar behavior using [callbacks](https://swagger.io/specification/#callbackObject): a callback defines a request that the server may initiate in response to receiving another request. Callbacks can thus be used to model publish/subscribe behavior. I.e., when the server receives a request to update some data, it can then itself issue callback requests (outside of the first request/response cycle) to any number of subscribed clients to inform about the new data.
+
+When enabling the `createSubscriptionsFromCallbacks` option, OpenAPI-to-GraphQL creates a subscription field for any operation in the given OpenAPI document that defines a `callback` object. In such cases, OpenAPI-to-GraphQL creates a [subscribe](http://spec.graphql.org/draft/#Subscribe()) function responsible to subscribing clients to receive results of callbacks being executed, and a special form of a [resolve](http://spec.graphql.org/draft/#ResolveFieldEventStream()) function, which pushes data updates to subscribed clients.
+
+To create these two functions, OpenAPI-to-GraphQL relies on the popular [graphql-subscriptions](https://github.com/apollographql/graphql-subscriptions) package, which provides a unified API to support different network transports (like WebSockets, MQTT, Redis etc. - see this [list of supported transports](https://github.com/apollographql/graphql-subscriptions#pubsub-implementations)).
+
+A typical example of using OpenAPI-to-GraphQL to create a GraphQL server supporting subscriptions may look like this:
+
+### Creating PubSub instance 
+
+First, initialize a PubSub instance to spread events between your API and the GraphQL Server, in a `pubsub.js` file.
+
+```javascript
+import { EventEmitter2 } from 'eventEmitter2';
```

Comment:
I don't understand why this needs to be "eventEmitter2". Is the 2 necessary?

Reply:
It's not necessary, but eventEmitter2 allows to use wildcard in the topic chain. Which is very useful when you compose a topic when you have variable field that cannot be determined in advanced, but which are part of the topic structure. I mentioned it in the SUBSCRIPTION doc.